tutor/commands/k8s.py

```diff
@@ -296,13 +296,20 @@ def start(context: K8sContext, names: List[str]) -> None:
     names = names or ["all"]
     for name in names:
         if name == "all":
-            # Create volumes
-            kubectl_apply(
-                context.root,
-                "--wait",
-                "--selector",
-                "app.kubernetes.io/component=volume",
-            )
+            volumes_manifest = tutor_env.pathjoin(context.root, "k8s/volumes.yml")
+            with open(volumes_manifest) as volumes_manifest_file:
```

Comment:
I'm sorry, I should have read this PR more thoroughly. Reading rendered environment files from tutor is really not a great idea (with some minor exceptions). Also, this quick fix would miss quite a few edge cases, such as volumes defined outside of volumes.yml.

Unfortunately I don't have a great and easy-to-implement alternative to suggest. Ideally, we would just ignore errors when command output includes the "error: no objects passed to apply" string. But there is no straightforward way to achieve this.

I explored alternatives, such as telling kubectl to ignore such errors. In my understanding, it's just not a good user API to crash on missing objects. Ideally, we should see only a warning. But I found no way to change the current behaviour.

Do you have alternative ideas?

Reply:
In my experience, this is failing because deployments that depend on volumes won't launch if the volumes don't exist already. But maybe this is no longer the case, or I'm hallucinating. Can you please try it out? I agree it would be a great way to fix this.

Reply:
I tested this approach and it worked quite well. Here's the log of `tutor k8s start`: 

```other
kubectl get namespaces openedx
Error from server (NotFound): namespaces "openedx" not found
Namespace does not exist: now creating it...
kubectl apply --kustomize /Users/snglth/Development/Abstract/test-tutor/env --wait --selector app.kubernetes.io/component=namespace
namespace/openedx created
kubectl apply --kustomize /Users/snglth/Development/Abstract/test-tutor/env --selector 'app.kubernetes.io/component notin (job,namespace)'
configmap/caddy-config-84d5m8f228 created
configmap/mfe-caddy-config-g4549gh2g6 created
configmap/openedx-config-6f26g64k46 created
configmap/openedx-settings-cms-7kb9hbbfg7 created
configmap/openedx-settings-lms-645456bbt8 created
configmap/openedx-uwsgi-config-f57k74k5cc created
configmap/redis-config-6588tm476h created
service/caddy created
service/cms created
service/elasticsearch created
service/forum created
service/lms created
service/mfe created
service/minio created
service/mongodb created
service/mysql created
service/redis created
service/smtp created
persistentvolumeclaim/caddy created
persistentvolumeclaim/elasticsearch created
persistentvolumeclaim/minio created
persistentvolumeclaim/mongodb created
persistentvolumeclaim/mysql created
persistentvolumeclaim/redis created
deployment.apps/caddy created
deployment.apps/cms created
deployment.apps/cms-worker created
deployment.apps/elasticsearch created
deployment.apps/forum created
deployment.apps/lms created
deployment.apps/lms-worker created
deployment.apps/mfe created
deployment.apps/minio created
deployment.apps/mongodb created
deployment.apps/mysql created
deployment.apps/redis created
deployment.apps/smtp created
```

Notice that `persistentvolumeclaim` objects are being created before deployments. I also pushed my changes.

Reply:
Thank. It's much cleaner this way. Can you please just add a changelog entry? https://docs.tutor.overhang.io/tutor.html#contributing